src/mpid/ch4/netmod/ofi/ofi_init.c

```diff
@@ -1366,12 +1366,25 @@ static int update_global_limits(struct fi_info *prov)
         MPIDI_OFI_global.max_msg_size = MPL_MIN(prov->ep_attr->max_msg_size, MPIR_AINT_MAX);
     }
     MPIDI_OFI_global.stripe_threshold = MPIR_CVAR_CH4_OFI_STRIPING_THRESHOLD;
-    MPIDI_OFI_global.max_order_raw = prov->ep_attr->max_order_raw_size;
-    MPIDI_OFI_global.max_order_war = prov->ep_attr->max_order_war_size;
-    MPIDI_OFI_global.max_order_waw = prov->ep_attr->max_order_waw_size;
-    MPIDI_OFI_global.tx_iov_limit = MIN(prov->tx_attr->iov_limit, MPIDI_OFI_IOV_MAX);
-    MPIDI_OFI_global.rx_iov_limit = MIN(prov->rx_attr->iov_limit, MPIDI_OFI_IOV_MAX);
-    MPIDI_OFI_global.rma_iov_limit = MIN(prov->tx_attr->rma_iov_limit, MPIDI_OFI_IOV_MAX);
+    if (prov->ep_attr->max_order_raw_size > MPIR_AINT_MAX) {
+        MPIDI_OFI_global.max_order_raw = MPIR_AINT_MAX;
+    } else {
+        MPIDI_OFI_global.max_order_raw = prov->ep_attr->max_order_raw_size;
+    }
+    if (prov->ep_attr->max_order_war_size > MPIR_AINT_MAX) {
+        MPIDI_OFI_global.max_order_war = MPIR_AINT_MAX;
+    } else {
+        MPIDI_OFI_global.max_order_war = prov->ep_attr->max_order_war_size;
+    }
+    if (prov->ep_attr->max_order_waw_size > MPIR_AINT_MAX) {
+        MPIDI_OFI_global.max_order_waw = MPIR_AINT_MAX;
+    } else {
+        MPIDI_OFI_global.max_order_waw = prov->ep_attr->max_order_waw_size;
+    }
+
+    MPIDI_OFI_global.tx_iov_limit = MPL_MIN(prov->tx_attr->iov_limit, MPIDI_OFI_IOV_MAX);
+    MPIDI_OFI_global.rx_iov_limit = MPL_MIN(prov->rx_attr->iov_limit, MPIDI_OFI_IOV_MAX);
+    MPIDI_OFI_global.rma_iov_limit = MPL_MIN(prov->tx_attr->rma_iov_limit, MPIDI_OFI_IOV_MAX);
     MPIDI_OFI_global.max_mr_key_size = prov->domain_attr->mr_key_size;
```

Comment:
Maybe better to separate `MIN`->`MPL_MIN` to different commit

Reply:
Separated.